packages/core/helper-plugin/src/hooks/useSelectionState/index.js

```diff
@@ -34,5 +34,31 @@ export const useSelectionState = (keys, initialValue) => {
     }
   };
 
-  return [selections, { selectOne, selectAll, selectOnly, setSelections }];
+  const selectMultiple = (nextSelections) => {
+    setSelections([
+      ...selections,
+      ...nextSelections.filter(
+        (nextSelection) =>
+          selections.findIndex((currentSelection) =>
+            keys.every((key) => currentSelection[key] === nextSelection[key])
+          ) === -1
+      ),
+    ]);
+  };
+
+  const deselectMultiple = (nextSelections) => {
+    setSelections([
+      ...selections.filter(
+        (currentSelection) =>
+          nextSelections.findIndex((nextSelection) =>
+            keys.every((key) => currentSelection[key] === nextSelection[key])
+          ) === -1
+      ),
+    ]);
+  };
```

Comment:
I find this code hard to read at the moment, can we please add notes to explain at a high level what we're trying to do with an example?

Reply:
I have added comments to the code for clarity. In summary, these two methods are responsible for handling the selection of multiple files without modifying the already selected items. Additionally, they should prevent duplicate entries in the array of selected items. Initially, I considered using the some method to check if an object exists in the array, but ultimately opted for findIndex to maintain consistency within this hook.